flyteadmin/pkg/repositories/transformers/task_execution.go

```diff
@@ -393,7 +419,11 @@ func UpdateTaskExecutionModel(ctx context.Context, request *admin.TaskExecutionE
 		reportedAt = request.Event.OccurredAt
 	}
 	taskExecutionClosure.UpdatedAt = reportedAt
-	taskExecutionClosure.Logs = mergeLogs(taskExecutionClosure.Logs, request.Event.Logs)
+
```

Comment:
On a high level, the contract between flyteplugins/flytepropeller and flyteadmin changes the following way:

*Before:*

* If a plugin wants a certain task log link to be shown from a certain phase onwards (almost exclusively the `Running` phase, only exception being the spark plugin), the plugin needs to send this log link in that phase.
    * Subsequently, the task log doesn't have to be sent again since flyteadmin aggregates the log links.

*After*:

* If a plugin wants a certain task log link to be shown from a certain phase onwards, the plugin needs to send this log link *at least* in that phase.
    * If task links are sent in an earlier phase, they will be ignored by admin
    * Subsequently, the task log still doesn't have to be sent again since flyteadmin still aggregates the log links.
    * Plugins can mark log links to be deleted once the execution finished.

---

Reply:
I did briefly consider whether a plugin should be able to send a log link which will e.g. be shown only from `Running` onwards only a single time already in a prior phase, e.g. `Pending`. This would mean that flyteadmin would have to track the still inactive log link and then make it active once the running phase is reached.

I personally think this is an overkill though since currently all plugins always send all links.